npc/custom/warper.txt

```diff
@@ -69,7 +70,19 @@ function Go {
 	set lastwarp$, getarg(0);
 	set lastwarpx, getarg(1,0);
 	set lastwarpy, getarg(2,0);
-	warp getarg(0),getarg(1,0),getarg(2,0);
+	for(.@i=0;.@i<getarraysize(.random_coordinates$);.@i++){
+		if(lastwarp$ == .random_coordinates$[.@i] && getarg(3,0) != 1){
+			set lastwarpx, 0;
+			set lastwarpy, 0;
+		}
+	}	
+	for(.@i=0;.@i<getarraysize(.blocked_maps$);.@i++){
```

Comment:
If the array’s size is static for every iteration, store the size in a variable instead.

Reply:
Same here, block them in the OnInit code, maybe using negative coordinates?

Reply:
here i am thinking if i can move this check to the Pick function so the blocked map wont be showing as an option for the players